iron-fit-behavior.html

```diff
@@ -505,40 +488,88 @@
       return target.getBoundingClientRect();
     },
 
-    __isAlignRight: function(hAlign, size, positionRect, fitRect) {
-      if (!hAlign || hAlign === 'auto') {
-        // We should align on the right if positionTarget is on the right of fitInto,
-        // or if we can overlap and aligning on the left would cause more cropping
-        // than aligning on the right.
-        var positionTargetCenter = positionRect.left + positionRect.width/2;
-        var fitIntoCenter = fitRect.left + fitRect.width/2;
-        var croppedLeft = Math.abs(Math.min(0, positionRect.left));
-        var croppedRight = Math.abs(Math.min(0, positionRect.right - size));
-        return positionTargetCenter > fitIntoCenter ||
-          (!this.noOverlap && croppedLeft > croppedRight);
-      }
-      return hAlign === 'right';
-    },
-
-    __isAlignBottom: function(vAlign, size, positionRect, fitRect) {
-      if (!vAlign || vAlign === 'auto') {
-        // We should align on the bottom if positionTarget is on the bottom of fitInto,
-        // or if we can overlap and aligning on the top would cause more cropping
-        // than aligning on the bottom.
-        var positionTargetCenter = positionRect.top + positionRect.height/2;
-        var fitIntoCenter = fitRect.top + fitRect.height/2;
-        var croppedTop = Math.abs(Math.min(0, positionRect.top));
-        var croppedBottom = Math.abs(Math.min(0, positionRect.bottom - size));
-        return positionTargetCenter > fitIntoCenter ||
-          (!this.noOverlap && croppedTop > croppedBottom);
-      }
-      return vAlign === 'bottom';
-    },
-
     __getCroppedArea: function(rect, fitRect) {
       var verticalCrop = Math.min(0, rect.top) + Math.min(0, fitRect.bottom - (rect.top + rect.height));
       var horizontalCrop = Math.min(0, rect.left) + Math.min(0, fitRect.right - (rect.left + rect.width));
       return Math.abs(verticalCrop) * rect.width + Math.abs(horizontalCrop) * rect.height;
+    },
+
+
+    __getPosition: function(hAlign, vAlign, size, positionRect, fitRect) {
+      // All the possible configurations.
+      // Ordered as top-left, top-right, bottom-left, bottom-right.
+      var positions = [{
+        verticalAlign: 'top',
+        horizontalAlign: 'left',
+        top: positionRect.top,
+        left: positionRect.left
+      }, {
+        verticalAlign: 'top',
+        horizontalAlign: 'right',
+        top: positionRect.top,
+        left: positionRect.right - size.width
+      }, {
+        verticalAlign: 'bottom',
+        horizontalAlign: 'left',
+        top: positionRect.bottom - size.height,
+        left: positionRect.left
+      }, {
+        verticalAlign: 'bottom',
+        horizontalAlign: 'right',
+        top: positionRect.bottom - size.height,
+        left: positionRect.right - size.width
+      }];
```

Comment:
I wonder if there is a benefit to making this a computed value for `size` and `positionRect` so that you don't need to re-allocate it for every call to `__getPosition`. WDYT?